bin/propolis-server/src/lib/spec/builder.rs

```diff
@@ -22,35 +39,45 @@ pub enum SpecBuilderError {
     PciPathInUse(PciPath),
 
     #[error("Serial port {0:?} is already specified")]
-    SerialPortInUse(components::devices::SerialPortNumber),
+    SerialPortInUse(SerialPortNumber),
 
+    #[cfg(feature = "falcon")]
```

Comment:
Hmm, this forces anyone who wants to match on a `SpecBuilderError` to have a "falcon" feature of their own --- I wonder if it would be better to not make the error variant conditional on "falcon", for that reason?  This would be fine to do, since the error doesn't include any falcon-only type... It could just be documented that it's only returned when falcon is enabled, so that consumers can match without having their own falcon feature flag.

On the other hand, I don't think there are presently any consumers who match on this, so it probably doesn't really matter. But, having a structured error enum makes it _possible_ for other code to do that.

Reply:
upon further consideration, i realize that this type is not actually a public API, even though it's `pub`...it's only exposed to the `propolis-server` binary, which has a "falcon" feature anyway. Never mind!

Reply:
I demoted it to `pub(crate)` anyway in 6b5ad3e to make the intent a little clearer here.